teos/src/api/internal.rs

```diff
@@ -380,25 +424,156 @@ mod tests_private_api {
         let response = internal_api
-            .get_all_appointments(Request::new(()))
+            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                locator: locator1.clone(),
+            }))
             .await
             .unwrap()
             .into_inner();
 
+        // The response should contain only one appointment with locator1.
         assert_eq!(response.appointments.len(), 1);
         assert!(matches!(
             response.appointments[0].appointment_data,
-            Some(msgs::appointment_data::AppointmentData::Tracker { .. })
+            Some(msgs::appointment_data::AppointmentData::Appointment(
+                msgs::Appointment {
+                    ref locator,
+                    ..
+                }
+            )) if *locator == locator1
+        ));
+
+        let response = internal_api
+            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                locator: locator2.clone(),
+            }))
+            .await
+            .unwrap()
+            .into_inner();
+
+        // The response should contain only one appointment with locator2.
+        assert_eq!(response.appointments.len(), 1);
+        assert!(matches!(
+            response.appointments.first().unwrap().appointment_data,
+            Some(msgs::appointment_data::AppointmentData::Appointment(
+                msgs::Appointment {
+                    ref locator,
+                    ..
+                }
+            )) if *locator == locator2
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_get_appointments_responder() {
+        let internal_api = create_api().await;
+
+        // Generate and add two random trackers to the responder.
+        let tracker1 = internal_api.watcher.add_random_tracker_to_responder(None);
+        let tracker2 = internal_api.watcher.add_random_tracker_to_responder(None);
+
+        // Calculate the locators for the generated trackers.
+        let locator1 = Locator::new(tracker1.dispute_tx.txid()).serialize();
+        let locator2 = Locator::new(tracker2.dispute_tx.txid()).serialize();
+
+        // Query for each locator and assert it retrieves the correct tracker.
+        let response = internal_api
+            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                locator: locator1.clone(),
+            }))
+            .await
+            .unwrap()
+            .into_inner();
+
+        // The response should contain only one tracker with dispute txid that matches locator1.
+        assert_eq!(response.appointments.len(), 1);
+        assert!(matches!(
+            response.appointments[0].appointment_data,
+            Some(msgs::appointment_data::AppointmentData::Tracker(
+                msgs::Tracker {
+                    ref dispute_txid,
+                    ..
+                }
+            )) if dispute_txid[..LOCATOR_LEN] == locator1
+        ));
```

Comment:
Once this is not duplicated, it can be simplified:

```suggestion
        // Calculate the locators for the generated trackers.
        let locator = Locator::new(tracker.dispute_tx.txid());

        // Query for each locator and assert it retrieves the correct tracker.
        let response = internal_api
            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
                locator: locator.serialize(),
            }))
            .await
            .unwrap()
            .into_inner();

        // The response should contain only one tracker with dispute txid that matches the current iteration.
        assert_eq!(response.appointments.len(), 1);
        assert!(matches!(
            response.appointments[0].appointment_data,
            Some(msgs::appointment_data::AppointmentData::Tracker(
                msgs::Tracker {
                    ref dispute_txid,
                    ..
                }
            )) if  Locator::deserialize(dispute_txid).unwrap() == locator
        ));
```

Reply:
Notice I've changed a bit the way the locators are compared, mainly for readability but also to abstract the actual generation. Locator generation should not be changed in the future, but it they happen to, this tests would need to be updated if using `dispute_txid[..LOCATOR_LEN]`. Using `deserialize` prevents that from happening.